### PR TITLE
[DO NOT MERGE] Revert Nemotron perf changes (#26733) — CI debug for release

### PR DESCRIPTION
**Pull request by b8zhong**
Debug-only PR for the Nemotron-3-Super NVFP4 + MTP NaN (`verify: target model logits`) on `release/v0.5.13`.

Full `git revert` of #26733 (`38ae22e08c`, *Nemotron perf changes*) — the suspected regression. Clean inverse, no conflicts: 15 files, +58/-297 (mirror of #26733's +297/-58). Undoes all of:
- routed_scaling_factor fusion (restores `final_hidden_states *= routed_scaling_factor`)
- gated-RMSNorm PDL/GDC
- mamba `output=None` refactor
- `relu2` jit kernel + `ReLU2.forward_cuda`
- router `torch.mm` precision change

This is the ground-truth bisect anchor: if this goes green, the NaN is definitively in #26733 (and the narrower PRs pin which hunk). If it still NaNs, the cause is *outside* #26733.

Isolation matrix (all off release/v0.5.13):
- #27810 — release as-is (baseline, expected NaN)
- #27812 — revert routed_scaling_factor fusion only
- #27814 — trtllm_mha full-attn + mamba extra_buffer
- this PR — full revert of #26733

**Do not merge.**

---
### CI States

Latest PR Test (Base): :x: [Run #27285655336](https://github.com/sgl-project/sglang/actions/runs/27285655336)
Latest PR Test (Extra): :x: [Run #27285655932](https://github.com/sgl-project/sglang/actions/runs/27285655932)

### CODE REVIEW

**Comment by b8zhong** [comment]
/rerun-test test/registered/8-gpu-models/test_nvidia_nemotron_3_super_nightly.py test/registered/4-gpu-models/test_nvidia_nemotron_3_super_nvfp4.py

**Comment by mmangkad** [comment]
I can't reproduce the failure on either SM100 or SM103 with the current `release/v0.5.13` branch

**Comment by b8zhong** [comment]
@mmangkad Try:

cd /sgl-workspace/sglang/test && SGLANG_SKIP_SGL_KERNEL_VERSION_CHECK=1 SGLANG_ENABLE_ASYNC_ASSERT=1 python registered/4-gpu-models/test_nvidia_nemotron_3_super_nvfp4.py TestNvidiaNemotron3SuperNVFP4MTP -v

(for some reason, launching the server on its own and sending some common requests doesn't reproduce this issue)
